tailscale/provider_test.go

```diff
@@ -140,11 +140,11 @@ func checkResourceRemoteProperties(resourceName string, fn func(client *tsclient
 		}
 
 		client := testAccProvider.Meta().(*tailscale.Clients).V2
-		return fn(client, rs)
+		return check(client, rs)
 	}
 }
 
-func checkResourceDestroyed[T any](resourceName string, get func(client *tsclient.Client, rs *terraform.ResourceState) (T, error)) resource.TestCheckFunc {
+func checkResourceDestroyed(resourceName string, check func(client *tsclient.Client, rs *terraform.ResourceState) error) resource.TestCheckFunc {
```

Comment:
I realized that depending on the test, our logic for checking differs, so I made this more generic.